.github/actions/check_artifact_size/action.yaml

```diff
@@ -0,0 +1,64 @@
+name: Check Artifact Size
+description: Check if the size of the newly created artifact specified by path is larger than the size of the previously uploaded artifact specified by name by the threshold.
```

Comment:
This check will only be enabled when `check_artifact_size` fields is set using the main workflow. See the [if condition](https://github.com/youtube/cobalt/pull/3813/files#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R272). 
Job will fail if the size check failed.

Reply:
All new checks are considered required by the [update script](https://github.com/youtube/cobalt/blob/main/tools/update_required_branch_checks.py).

If we want this to be an optional check it would have to be added to the exclude list.

Reply:
Taking a step back from the implementation, these are the behaviors I think we want:
* The size check failure should be brought to the attention of the PR author and reviewers. This is so that they can explore whether the change can be made more efficiently w.r.t. binary size.
* If the authors and reviewers decide that a PR should still be merged despite the size check failure, they should be able to do that. (It would be nice if we could enforce acknowledgement of this in the PR using tooling - for example,  a comment justifying it - but that's "nice to have" and not a requirement.)
* If PRs are still merged while failing the check, we want them to be discoverable. It should be easy for someone (e.g., the build cop) to look back and find all PRs merged over the last xx that failed the size check at merge time.

@haozheng-cobalt @oxve do you have a suggestion for what would work best to accomplish this?

Reply:
On just one point, discoverability: auto-labeling the PR works. It's easily discoverable in Github UI, we already use a workflow to do that for "outside collaborator". So we could just label the PRs with "exceeded size check" or such.

Reply:
Can you also add the name of this check to the exclude list?

https://github.com/youtube/cobalt/blob/main/tools/update_required_branch_checks.py#L32

Reply:
quoted from Oscar: "Now that the check is part of the main workflow you shouldn't do this. This would only be relevant if it was its own workflow."